whisk/client.go

```diff
@@ -214,19 +231,26 @@ func (c *Client) LoadX509KeyPair() error {
 		}
 	}
 
-	// Use the defaultTransport as the transport basis to maintain proxy support
-	c.client.Transport = &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
-		DialContext: (&net.Dialer{
-			Timeout:   30 * time.Second,
-			KeepAlive: 30 * time.Second,
-			DualStack: true,
-		}).DialContext,
-		MaxIdleConns:          100,
-		IdleConnTimeout:       90 * time.Second,
-		TLSHandshakeTimeout:   10 * time.Second,
-		ExpectContinueTimeout: 1 * time.Second,
-		TLSClientConfig:       tlsConfig,
+	// only set new transport when the TLS config is nondefault
+	if tlsModified {
```

Comment:
maybe no need to introduce a new var (tlsModified) that's set in two places.. instead just
```
// Only replace the existing transport when a custom TLS configuration is needed
if c.Config.Insecure || tlsConfig.Certificates != nil {
```

Reply:
even better!